### DISCUSSION TITLE
How to give custom marks higher priority?

### DISCUSSION THREAD

**Discussion by chrisgrieser** [petertriho/nvim-scrollbar#62] [Q&A]
I have created this simple custom mark that solely displays the current line. However, I am not sure what I have to do to make the "current line" mark override the other marks, e.g. from gitsign.

```lua
require("scrollbar.handlers").register("currentLine", function(bufnr)
	local currentLnum = fn.line(".")
	return {{ line = currentLnum, text = "", type = "Hint", level = 0}}
end)
```

I tried `level = 6` and `level = 0` (assuming those levels correspond to priority from the config), but with both of those, the gitsign marks take priority over my custom mark.

**Comment by petertriho** [comment]
Sorry, I really need to improve the docs on this. Level is for "text" index used under `config.marks.[type].text`
To set a higher priority, you are using "Hint" there, so in the setup:

```lua
require("scrollbar").setup({
  marks = {
    Hint = {
      priority = 0,
    }
  }
})

**Reply by chrisgrieser** [reply]
I tried setting the priority to `0` as well as `6`, in both cases my custom mark is overridden by any other mark. I suppose that's a bug or something?

**Reply by petertriho** [reply]
Ah sorry, it is a lower number (so 0), but I suspect it's because the cursor mark is only being rendered when something else (like lsp) is rendering the scrollbar that the mark shows up. You should try and use autocmd events to render the cursor

**Reply by petertriho** [reply]
I really should just add this to the included handlers

Example:
```lua
        require("scrollbar").setup({
            marks = {
                Cursor = { text = "", priority = -1, highlight = "Normal" },
            },
        })

        vim.api.nvim_create_autocmd({ "CursorHold", "CursorHoldI", "CursorMoved", "CursorMovedI" }, {
            callback = function()
                local current_line = vim.fn.line(".") - 1
                local bufnr = vim.fn.bufnr("%")

                local utils = require("scrollbar.utils")
                local scrollbar_marks = utils.get_scrollbar_marks(bufnr)
                scrollbar_marks.cursor = { { line = current_line, text = "", type = "Cursor" } }
                utils.set_scrollbar_marks(bufnr, scrollbar_marks)

                require("scrollbar").render()
            end
        })

```

**Reply by petertriho** [reply]
Priority should be fixed now https://github.com/petertriho/nvim-scrollbar/commit/819de9b6f7a43de0840d6529a81d4ee1e10a6330

**Reply by petertriho** [reply]
Cursor handler now included

**Reply by chrisgrieser** [reply]
Fantastic, thank you! Works as expected
